### ISSUE TITLE
Docker build fails

### ISSUE THREAD

**Issue by dudeperf3ct** [gchq/Bailo#143]
### Subject of the issue

I am following the instructions mentioned in [Helm chart instructions](https://github.com/gchq/Bailo/tree/main/helm). The [configuration step](https://github.com/gchq/Bailo/tree/main/helm#configuration) mentions creating and hosting Bailo image. The  image can be built using `docker build -t bailo .` from root directory of project.

This step fails to build a docker image.

### Your environment

- version / commit hash of bailo (`git rev-parse HEAD`) : main branch
- runtime environment (`docker-compose`, `openshift`) : kubectl / helm

### Steps to reproduce

Running `docker build -t bailo .` from root directory.

Error

```bash
unable to convert uid/gid chown string to host mapping: can't find gid for group nextjs: no such group: nextjs
```

[COPY command](https://docs.docker.com/engine/reference/builder/#copy) in docker recommends using `--chown=<user>:<group>`. Replacing `--chown=nextjs:nextjs`with `--chown=nextjs:nodejs`, the image build is a success.

Running the container, I get the following error

```bash
 bailo@0.3.0 start
> cross-env NODE_ENV=production node dist/server/index.js

node:internal/modules/cjs/loader:959
  throw err;
  ^

Error: Cannot find module '../../lib/p-mongo-queue/pMongoQueue'
Require stack:
- /app/dist/server/utils/queues.js
- /app/dist/server/processors/processDeployments.js
- /app/dist/server/index.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/app/dist/server/utils/queues.js:10:39)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/app/dist/server/processors/processDeployments.js:13:18)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.Module._load (node:internal/modules/cjs/loader:839:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/app/dist/server/utils/queues.js',
    '/app/dist/server/processors/processDeployments.js',
    '/app/dist/server/index.js'
  ]
}
npm notice 
npm notice New minor version of npm available! 8.15.0 -> 8.19.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v8.19.2
npm notice Run npm install -g npm@8.19.2 to update!
npm notice 
```

On comparing `Dockerfile` with `Dockerfile.prod`, I find that the following line is missing

```dockerfile
COPY --from=builder --chown=nextjs:nextjs /app/lib/p-mongo-queue ./dist/lib/p-mongo-queue
```

Adding the above line with the right group, the build is a success.

Pushing this image and running the instructions as provided in [helm chart instruction](https://github.com/gchq/Bailo/tree/main/helm), the pod crashes with `CrashLoopBackOff` error.

Pod Events

```bash
vents:
  Type     Reason     Age                     From               Message
  ----     ------     ----                    ----               -------
  Normal   Scheduled  3m53s                   default-scheduler  Successfully assigned bailo/bailo-6b995bcf9-8g78v to bailo-control-plane
  Normal   Created    3m9s (x4 over 3m52s)    kubelet            Created container bailo
  Normal   Started    3m9s (x4 over 3m52s)    kubelet            Started container bailo
  Warning  BackOff    2m36s (x12 over 3m50s)  kubelet            Back-off restarting failed container
  Normal   Pulled     2m25s (x5 over 3m52s)   kubelet            Container image "dudeperf3ct7/bailo:latest" already present on machine
```

Tailing log for the pod provides following information

```bash
Failed to open temporary file /etc/ssl/certs/bundleXXXXXX for ca bundle
No schemas setup script set; You will have to add schemas after startup
Running 'npm run start'
```

### Expected behaviour

Everything should work. i.e. just building the image and adding that to helm should give access to Model UI at 3000 port.

**Comment by a3957273** [comment]
Closed in #144

**Comment by dudeperf3ct** [comment]
@a3957273, I followed the helm instructions and pushed a Bailo image (created from `Dockerfile.prod`) to docker hub. Upon writing a custom `local.yaml` file for overriding the Bailo image and running the further instructions, I still get the `CrashLoopBackOff` error (Back-off restarting failed container). 

How can I resolve this error? I have provided detailed information in the last section of the issue.
